gym_graph_traffic/envs/intersection.py

```diff
@@ -117,3 +121,1003 @@ def can_i_go(self, from_idx) -> int:
     def pass_car(self, from_idx, car_position, car_velocity) -> None:
         (_, dest) = self.dest_dict[from_idx]
         dest.new_car_at = (car_position, car_velocity)
+
+
+class FourWayTurnsIntersection(Intersection):
```

Comment:
So this class can probably be implemented by considering one case and rotating the intersection matrix and some other properties to avoid code duplication. You will probably the main user of this code, so it can stay as it is if you want, but you might stumble upon some bugs in the future, that's your call. You might want to add some "TODO" comment as a reminder it can be refactored in the future.
Another thing that we might want to change is the function names and intersection api in general, but that's a topic for a different discussion and pull request.